packer/plugin-getter/plugins.go

```diff
@@ -105,84 +109,103 @@ func (pr Requirement) ListInstallations(opts ListInstallationsOptions) (InstallL
+	}
+	for _, path := range matches {
+		fname := filepath.Base(path)
+		if fname == "." {
+			continue
+		}
+
+		// base name could look like packer-plugin-amazon_v1.2.3_x5.1_darwin_amd64.exe
+		versionsStr := strings.TrimPrefix(fname, FilenamePrefix)
+		versionsStr = strings.TrimSuffix(versionsStr, filenameSuffix)
+
 		if pr.Identifier == nil {
-			glob = filepath.Join(knownFolder, "*", "*", "*", FilenamePrefix+"*"+filenameSuffix)
-		} else {
-			glob = filepath.Join(knownFolder, pr.Identifier.Hostname, pr.Identifier.Namespace, pr.Identifier.Type, FilenamePrefix+"*"+filenameSuffix)
+			if idx := strings.Index(versionsStr, "_"); idx > 0 {
+				versionsStr = versionsStr[idx+1:]
+			}
 		}
 
-		matches, err := filepath.Glob(glob)
+		descOut, err := exec.Command(path, "describe").Output()
 		if err != nil {
-			return nil, fmt.Errorf("ListInstallations: %q failed to list binaries in folder: %v", pr.Identifier.String(), err)
+			log.Printf("couldn't call describe on %q, ignoring", path)
+			continue
 		}
-		for _, path := range matches {
-			fname := filepath.Base(path)
-			if fname == "." {
-				continue
-			}
 
-			// base name could look like packer-plugin-amazon_v1.2.3_x5.1_darwin_amd64.exe
-			versionsStr := strings.TrimPrefix(fname, FilenamePrefix)
-			versionsStr = strings.TrimSuffix(versionsStr, filenameSuffix)
-
-			if pr.Identifier == nil {
-				if idx := strings.Index(versionsStr, "_"); idx > 0 {
-					versionsStr = versionsStr[idx+1:]
-				}
-			}
+		var describeInfo pluginsdk.SetDescription
+		err = json.Unmarshal(descOut, &describeInfo)
+		if err != nil {
+			log.Printf("%q: describe output deserialization error %q, ignoring", path, err)
+		}
 
-			// versionsStr now looks like v1.2.3_x5.1 or amazon_v1.2.3_x5.1
-			parts := strings.SplitN(versionsStr, "_", 2)
-			pluginVersionStr, protocolVerionStr := parts[0], parts[1]
-			pv, err := version.NewVersion(pluginVersionStr)
-			if err != nil {
-				// could not be parsed, ignoring the file
-				log.Printf("found %q with an incorrect %q version, ignoring it. %v", path, pluginVersionStr, err)
-				continue
-			}
+		// versionsStr now looks like v1.2.3_x5.1 or amazon_v1.2.3_x5.1
+		parts := strings.SplitN(versionsStr, "_", 2)
+		pluginVersionStr, protocolVerionStr := parts[0], parts[1]
+		pv, err := version.NewVersion(pluginVersionStr)
+		if err != nil {
+			// could not be parsed, ignoring the file
+			log.Printf("found %q with an incorrect %q version, ignoring it. %v", path, pluginVersionStr, err)
+			continue
+		}
 
-			// no constraint means always pass, this will happen for implicit
-			// plugin requirements and when we list all plugins.
-			if !pr.VersionConstraints.Check(pv) {
-				log.Printf("[TRACE] version %q of file %q does not match constraint %q", pluginVersionStr, path, pr.VersionConstraints.String())
-				continue
-			}
+		if strings.Replace(pluginVersionStr, "v", "", -1) != describeInfo.Version {
```

Comment:
Not that it is important here but if we find ourselves doing more of these version checks hashicorp/go-version supports checking if "v1.0.0" is equal to "1.0.0". It essentially drops the v prefix when comparing using the Equal function. 

This is a single usage so probably overkill but calling out in case we need to do this more.

Reply:
Yeah about this one: fun-fact, this changes in the follow-up PR, since the replace makes the versions invalid when working with `vM.m.p-dev`, as they end up being `M.m.p-de` :D